### ISSUE TITLE
Cannot find ovmf firmware files under openSUSE Tumbleweed host

### ISSUE THREAD

**Issue by FroggyFlox** [quickemu-project/quickemu#288]
#### Expected behaviour

Using an openSUSE (Tumbleweed) host, the following is expected to run successfully:
```shell
quickget ubuntu focal
quickemu --vm ubuntu-focal.conf --display gtk
```

#### Actual behaviour

`quickemu` fails to find the EFI firmware files despite them being correctly installed. To verify:
```shell
$ sudo zypper info qemu-ovmf-x86_64
[sudo] password for root: 
Loading repository data...
Reading installed packages...


Information for package qemu-ovmf-x86_64:
-----------------------------------------
Repository     : openSUSE-Tumbleweed-Oss
Name           : qemu-ovmf-x86_64
Version        : 202108-4.1
Arch           : noarch
Vendor         : openSUSE
Installed Size : 59.0 MiB
Installed      : Yes (automatically)
Status         : out-of-date (version 202108-2.1 installed)
Source package : ovmf-202108-4.1.src
Summary        : Open Virtual Machine Firmware - QEMU rom images (x86_64)
Description    : 
    The Open Virtual Machine Firmware (OVMF) project aims to support
    firmware for Virtual Machines using the edk2 code base.

    This package contains UEFI rom images for exercising UEFI secure
    boot in a qemu environment (x86_64)
```

Notably, the firmware files are installed under:
```shell
rpm -ql qemu-ovmf-x86_64
(...)
/usr/share/qemu/ovmf-x86_64-4m-code.bin
/usr/share/qemu/ovmf-x86_64-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-4m.bin
/usr/share/qemu/ovmf-x86_64-code.bin
/usr/share/qemu/ovmf-x86_64-ms-4m-code.bin
/usr/share/qemu/ovmf-x86_64-ms-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-ms-4m.bin
/usr/share/qemu/ovmf-x86_64-ms-code.bin
/usr/share/qemu/ovmf-x86_64-ms-vars.bin
/usr/share/qemu/ovmf-x86_64-ms.bin
/usr/share/qemu/ovmf-x86_64-opensuse-4m-code.bin
/usr/share/qemu/ovmf-x86_64-opensuse-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-opensuse-4m.bin
/usr/share/qemu/ovmf-x86_64-opensuse-code.bin
/usr/share/qemu/ovmf-x86_64-opensuse-vars.bin
/usr/share/qemu/ovmf-x86_64-opensuse.bin
/usr/share/qemu/ovmf-x86_64-smm-code.bin
/usr/share/qemu/ovmf-x86_64-smm-ms-code.bin
/usr/share/qemu/ovmf-x86_64-smm-ms-vars.bin
/usr/share/qemu/ovmf-x86_64-smm-ms.bin
/usr/share/qemu/ovmf-x86_64-smm-opensuse-code.bin
/usr/share/qemu/ovmf-x86_64-smm-opensuse-vars.bin
/usr/share/qemu/ovmf-x86_64-smm-opensuse.bin
/usr/share/qemu/ovmf-x86_64-smm-suse-code.bin
/usr/share/qemu/ovmf-x86_64-smm-suse-vars.bin
/usr/share/qemu/ovmf-x86_64-smm-suse.bin
/usr/share/qemu/ovmf-x86_64-smm-vars.bin
/usr/share/qemu/ovmf-x86_64-smm.bin
/usr/share/qemu/ovmf-x86_64-suse-4m-code.bin
/usr/share/qemu/ovmf-x86_64-suse-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-suse-4m.bin
/usr/share/qemu/ovmf-x86_64-suse-code.bin
/usr/share/qemu/ovmf-x86_64-suse-vars.bin
/usr/share/qemu/ovmf-x86_64-suse.bin
/usr/share/qemu/ovmf-x86_64-vars.bin
/usr/share/qemu/ovmf-x86_64-xen-4m.bin
/usr/share/qemu/ovmf-x86_64-xen.bin
/usr/share/qemu/ovmf-x86_64.bin
```

I noticed that `quickemu` is looking for these files in different locations:
https://github.com/quickemu-project/quickemu/blob/12fbf6b02f43dfa42be33f8da72bbdab50ac7667/quickemu#L386-L402

Is it simply because I have the "wrong" dependencies installed or missing, or is it due to different locations used by openSUSE Tumbleweed?

For information, I modifed `quickemu` to check for these files in `/usr/share/qemu/` and the error goes away and I can successfully run vms using `quickemu`:
(adding the following in line 398):
```shell
        elif [ -e "/usr/share/qemu/ovmf-x86_64-smm-ms-code.bin" ]; then
	        EFI_CODE="/usr/share/qemu/ovmf-x86_64-smm-ms-code.bin"
	        efi_vars "/usr/share/qemu/ovmf-x86_64-smm-ms-vars.bin" "${EFI_VARS}"
        else
```

#### Steps to reproduce the behaviour

```shell
quickget ubuntu focal
quickemu --vm ubuntu-focal.conf --display gtk
```

#### Quickemu output

```shell
$ ./quickemu --vm ubuntu-focal.conf --display gtk
Quickemu 3.11 using /usr/bin/qemu-system-x86_64 v6.1.0
 - Host:     "openSUSE Tumbleweed" running Linux 5.15 (T14s)
 - CPU:      AMD Ryzen 7 PRO 5850U with Radeon Graphics
 - CPU VM:   1 Socket(s), 4 Core(s), 2 Thread(s), 8G RAM
ERROR! EFI boot requested but no EFI firmware found.
       Please install OVMF firmware.
```

#### Linux Distribution & Kernel

Run the following and paste the output.

```bash
lsb_release --all
uname -a
```
```
LSB Version:    n/a
Distributor ID: openSUSE
Description:    openSUSE Tumbleweed
Release:        20211202
Codename:       n/a
Linux T14s 5.15.5-1-default #1 SMP Thu Nov 25 09:36:40 UTC 2021 (83fc974) x86_64 x86_64 x86_64 GNU/Linux
```

Thanks for any input!

**Comment by averyfreeman** [comment]
> Is it simply because I have the "wrong" dependencies installed or missing, or is it due to different locations used by openSUSE Tumbleweed?
> 
> For information, I modifed `quickemu` to check for these files in `/usr/share/qemu/` and the error goes away and I can successfully run vms using `quickemu`: (adding the following in line 398):

I wasn't sure these were the same type of files, as the `.fd` extension threw me off, so I copied the `OVMF_CODE.fd` etc. files I had from the OSX-KVM project to `/usr/share/OVMF`.  

But, having not read the spec yet, I imagine the `.fd` files are probably essentially the same under the hood, the `.fd` extension to denote that they are intended to stay under a certain size.

I think your adding a `if elif` is probably the easiest fix, but I was also thinking maybe having a check at the start of the script for distribution using `lsb_release -si` and then an array of variables for the default location of `ovmf` installations for each distro might be more elegant, as those long `if elif` banks are a bit cludgy looking, especially for tacking on even more conditions.

In any event, you should submit a PR.  If you're interested, there's a more active fork here: [https://github.com/TikoTikTok/quickemu](https://github.com/TikoTikTok/quickemu)

**Comment by flexiondotorg** [comment]
Closed via #285
